go/logictest/runner.go

```diff
@@ -128,7 +134,10 @@ func generateTestFile(harness Harness, f string) {
 	}()
 
 	for _, record := range testRecords {
-		schema, records, _, err := executeRecord(harness, record)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		lockCtx := context.WithValue(ctx, "lock", &loggingLock{ mux: &sync.Mutex{}, logged: false})
+
+		schema, records, _, err := executeRecord(lockCtx, cancel, harness, record)
```

Comment:
This method has to keep this lock stuff since the `logResult` function tries to get the lock from the context... Is this ok?

Reply:
yup totally fine